docs/design/rosbag2_playback_time.md

```diff
@@ -187,6 +187,53 @@ No special allowance needs to be made in the `Player` - time will be provided at
 
 The `Player` must register a `JumpHandler` with the `Clock` - so that when a jump occurs, the current message playback queue can be invalidated and re-enqueued according to the new starting time.
 
+## Time Synchronization and Conversion
+
+When implementing this time control, there are two separate time streams to consider:
+1. "ROS time": this is the time that is related to the messages in the bag - this time may speed up, slow down, pause, or jump based on rosbag playback controls (or external Time Sources)
+2. "Steady time": this time always moves forward, and always moves at the same rate; it is the real time that the user watching a playback experiences
+
+To implement `now() -> ROSTime` and `sleep_until(ROSTime until)`, we need to perform arbitrary conversion between "ROS Time" and "Steady Time".
```

Comment:
Every time we call `ROSTime PlayerClock::now()` we need to convert from Steady Time to ROS Time. Steady Time "now" is the one piece of information we always have available to us (via `std::chrono::steady_clock::now()`).

It is true at this moment that `Player` does not use `PlayerClock::now()`, it only uses `PlayerClock::sleep_until()`, but that doesn't mean that it will never be used. This functionality will be used to publish to `/clock` - which is a PR I already have in draft https://github.com/ros2/rosbag2/pull/695